metomi/rose/checksum.py

```diff
@@ -70,8 +70,16 @@ def get_checksum(name, checksum_func=None):
             for filename in filenames:
                 filepath = os.path.join(path, filename)
                 source = os.path.join(name, filepath)
-                checksum = checksum_func(source, name)
-                mode = os.stat(os.path.realpath(source)).st_mode
+
+                if os.path.islink(source) and not os.path.exists(source):
+                    # If the source is a broken link within a directory
+                    # install without failing (unlike a single file)
+                    checksum = os.path.realpath(source)
+                    mode = os.lstat(source).st_mode
+                else:
+                    checksum = checksum_func(source, name)
+                    mode = os.stat(os.path.realpath(source)).st_mode
```

Comment:
>  When installing directories using Git or Rsync broken symlinks will be copied.

This change targets the `get_checksum` function in general, not the git / rsync methods specificity.

Are we sure this method is only called for these methods, or should we pass the method name through to this function to ensure this logic is only activated when intended?

Also, this impacts Git, does it also impact SVN?

Reply:
> Are we sure this method is only called for these methods, or should we pass the method name through to this function to ensure this logic is only activated when intended?

I think that it makes sense to be consistent. The only other place this logic is used is Rose arch. I think it makes sense there too.

> Also, this impacts Git, does it also impact SVN?

I hope so - I would have tested it (and still can if you insist) but setting up the test would have taken extra time.

@dpmatthews  - what do you think?

Reply:
@oliver-sanders  please resolve if you're happy...